server/pbenchinacan/container-build.sh

```diff
@@ -118,12 +116,13 @@ buildah run $container chown --recursive pbench:pbench /srv/pbench
 # buildah run $container semanage fcontext -a -t httpd_sys_content_t /srv/pbench/public_html/static
 # buildah run $container restorecon -v -r /srv/pbench/archive /srv/pbench/public_html
 
-buildah run $container crontab -u pbench ${SERVER_LIB}/crontab/crontab
-
 sed -e "s/localhost/${HOSTNAME_F}/" ${PBINC_SERVER}/lib/config/pbench.httpd.conf >/tmp/pbench.conf.${$}
 buildah copy --chown root:root --chmod 0644 $container \
     /tmp/pbench.conf.${$} /etc/httpd/conf.d/pbench.conf
 rm /tmp/pbench.conf.${$}
+# We need to ensure HTTPD listens on port 8080 so container can be optionally
+# run with host networking from a non-root user.
+buildah run $container bash -c "sed -i -e 's/^Listen 80$/Listen 8080/' /etc/httpd/conf/httpd.conf"
```

Comment:
This seems weird:  it should be fine for HTTPD to use 80 inside the container -- when the container is invoked, we can map inside-port 80 to 8080 on the outside, even if the container is run from a non-root user and even if it is run with host networking.

Is there really a need to force things at this level?  (Wouldn't it be better to leave HTTPD configured to use port 80, which everyone expects, and then to remap it only when it's deployed?)

Reply:
Separate from the world of containers, one can't listen on port 80 without being root or at least being granted a capability (see e.g. https://superuser.com/questions/710253/allow-non-root-process-to-bind-to-port-80-and-443).

In the world of containers, when non-root user runs a container, it can map any unprivileged port on the host to ANY port in the container.

But when a non-root user runs a container using host networking, the process in the container is also non-root, and so can't bind to a privileged port.

So in the end, we want to allow a deployment that (for one reason or another) runs the container with host networking.

Reply:
I'll update the commit message for this commit to say that explicitly.

Reply:
I'm still not getting it.  🙁  If I run a container, I can map the port used inside the container to a different port outside the container.  So, I can map inside-port 80 to outside-port 8080; then, if I'm using host networking, the port that will be listened on will be unpriv'd, 8080.  Are you saying that my invocation _inside_ the container will fail, because it's trying to listen on a priv'd port?  I can see how that would be true if I the invocation there were made from a non-default, non-priv'd user account (e.g., `pbench`).  So, which problem are you trying to solve?

Reply:
> then, if I'm using host networking, the port that will be listened on will be unpriv'd, 8080

I believe with host networking the mappings are ignored.

Reply:
I have pulled the port change out of this PR so that we can work on the implications of that separate from these changes.  See PR #3108.

Reply:
> I believe with host networking the mappings are ignored.

Ah, so they are.  😝  I guess this is one "small" difference between Podman and Docker....

By default, rootless operation uses a different approach, but I imagine that it doesn't do what you want for which you've had to resort to host networking.